packages/backend/src/document.rs

```diff
@@ -123,3 +126,116 @@ pub struct RefContent {
     pub ref_id: Uuid,
     pub content: Value,
 }
+
+/// A subset of user relevant information about a ref. Used for showing
+/// users information on a variety of refs without having to load whole
+/// refs.
+#[derive(Clone, Debug, Serialize, Deserialize, TS)]
+pub struct RefStub {
+    pub name: String,
+    pub type_name: String,
+    pub ref_id: Uuid,
```

Comment:
Though it requires a little extra boilerplate, we should rename the fields in `RefStub` and `RefQueryParams` to match TypeScript case conventions where appropriate. E.g., this field should become:

```rust
#[serde(rename = "refId")]
pub ref_id: Uuid,
```

Reply:
done!